### ISSUE TITLE
Vehicle types with different capacity and price

### ISSUE THREAD

**Issue by teemusanteri** [google/or-tools#1316]
Hi,

Thank you for this great component.

Is it possible to optimize should we use just the truck or truck with trailer. Options are:

1. Truck, capacity 10tn, price 1,15€/km
2. Truck with the trailer, capacity 30tn, price 1,50€/km

I need to be able to select what type it is cost efficient to use. I tried to have vehicles so that I double the number of trucks and set capacities and distance matrix using the prices based on that but I don't have idea how to set constraint so that both versions of vehicles is not selected:

So basically I should be able to set constraint for example for 2 vehicles so that:
v1 = Truck 1. without trailer
v2 = Truck 1. with the trailer
v3 = Truck 2. without trailer
v4 = Truck 2. with the trailer

solver.add(v1 + v2 <= 1);
solver.add(v3 + v4 <= 1);

Is this even possible? How can it be done?

**Comment by Mizux** [comment]
cross post and answered on ortools-discuss: https://groups.google.com/g/or-tools-discuss/c/78xXgmsH6_Q/m/o6hs_uy6CAAJ
